extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1714,6 +1755,8 @@ private void checkCurrentTaskState() throws ExecutionException, InterruptedExcep
           continue;
         }
 
+        Preconditions.checkNotNull(taskData.status, "task[%s] has a null status", taskId);
```

Comment:
Similar question here -- when will this get thrown and what will happen when it gets thrown?

Reply:
Same here. This should never happen. NPE would throw if `taskData.status` is null and this is just a sanity code to see what's null on the potential NPE. I improved the error message.

Reply:
Thanks.